### PR TITLE
Switch to monorepo

### PR DESCRIPTION
**Pull request by ndelangen**
Issue: #742 

## What I did
Moved all the source for this package into a /packages/react-storybook
Merged in the monorepo master
Resolved mergeconflicts

## How to test
install dependencies - this should bootstrap lerna.
run `npm run test` in the root

### CODE REVIEW

**Comment by ndelangen** [comment]
Don't bother reviewing the code, review the idea of merging this, and deprecating all other repo's.
When you think we're ready.

File: packages/addons/packages.json [hunk 1]
```diff
@@ -0,0 +1,28 @@
+{
```
Comment (eliperelman) [inline packages/addons/packages.json]: This file is named `packages.json`, but most likely was meant to be `package.json`.

**Comment by tmeasday** [comment]
I personally would advocate doing it sooner (i.e. ASAP) and dealing with the issues as they come up; it's harder to anticipate them ahead of time. Unless we think there might be an issue that means the whole thing was a bad idea :)

**Comment by ndelangen** [comment]
@tmeasday If no-one objects I'll be merging this tomorrow?

**Comment by UsulPro** [comment]
> Unless we think there might be an issue that means the whole thing was a bad idea

@tmeasday  maybe worth to create a list of pros and cons? It might make the discussion more lively

> If no-one objects I'll be merging this tomorrow?

@ndelangen  what are the next steps after that? are we going to publish any beta version? what should we do with open PR's?

**Comment by ndelangen** [comment]
Unless PR's change package.json or readme, they should still be mergable!

Yeah, I think next step would be to trial run it on local existing storybook installations, see what works and what doesn't (the webpack 2 migration is in this). And we can also start publishing alpha versions.

**Comment by shilman** [comment]
@ndelangen Here are some notes from running it on a local project. TLDR; worked for me after a few minor bumps. Looks good!

1. When I initially installed using `npm install`, I saw this warning:
> Lerna v2.0.0-rc.1
> Encountered a cycle in the dependency graph. This may cause instability!
I don't think this is a serious issue, but I wanted to raise it in case you hadn't seen it.

2. Got the error `Error: Couldn't find preset "stage-0" relative to directory ...` when I tried to replace the version of storybooks I had with the current PR. Solved by adding `babel-preset-stage-0`.

3. Broke some of my private addons, but when I commented those out, everything worked fine. I'll upgrade the addons once this version has been released!

**Review by shilman — APPROVED** [review]
Looks good! See my comments above, and also I agree with @tmeasday about merging early.

**Comment by ndelangen** [comment]
> Encountered a cycle in the dependency graph. This may cause instability!
This message is a known issue: https://github.com/storybooks/mono/issues/19

File: .eslintrc [hunk 2]
```diff
@@ -1,27 +1,29 @@
...
+    "es6": true,
+    "node": true
+  },
+  "rules": {
+    "strict": 0,
+    "prettier/prettier": ["warn", {
+      "printWidth": 100,
+      "tabWidth": 2,
+      "bracketSpacing": true,
+      # "trailingComma": "es5",
```
Comment (aaronmcadam) [inline .eslintrc]: Do we want to decide on this comma rule? I vote for `all` as a value

File: .eslintrc [hunk 3]
```diff
@@ -1,27 +1,29 @@
...
+      "singleQuote": true
+    }],
+    "quotes": ["warn", "single"],
+    "arrow-parens": ["warn", "as-needed"]
+    # "comma-dangle": ["warn", "always"]
```
Comment (aaronmcadam) [inline .eslintrc]: Do we want to remove this line?

**Review by aaronmcadam — APPROVED** [review]

File: .eslintrc (hunk 3 above)
Comment (ndelangen) [inline .eslintrc]: done

File: .eslintrc (hunk 2 above)
Comment (ndelangen) [inline .eslintrc]: done

**Review by usulpro — APPROVED** [review]
+1 for mergin

**Pull request merged**